pkg/liqo-controller-manager/shadowpod-controller/shadowpod_controller.go

```diff
@@ -59,50 +67,78 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, err
 	}
 
-	pod := corev1.Pod{
+	// update shadowpod labels to include the "managed-by"
+	shadowPod.SetLabels(labels.Merge(shadowPod.Labels, labels.Set{consts.ManagedByLabelKey: consts.ManagedByShadowPodValue}))
+
+	// if any existing pod is already been created from the shadowpod...
+	existingPod := corev1.Pod{}
+	if err := r.Get(ctx, nsName, &existingPod); err != nil {
+		klog.Error("unable to get pod %q: %v", klog.KObj(&existingPod), err)
+	} else {
+		// ...Update existing pod or skip
+		if podShouldBeUpdated(&shadowPod, &existingPod) {
+			// Update Labels and Annotations
+			klog.V(4).Infof("pod %q found running, will update it with labels: %v and annotations: %v",
+				klog.KObj(&existingPod), existingPod.Labels, existingPod.Annotations)
+
+			// Create Apply object for Existing Pod
+			apply := corev1apply.Pod(existingPod.Name, existingPod.Namespace).
+				WithLabels(shadowPod.GetLabels()).
+				WithAnnotations(shadowPod.GetAnnotations())
+
+			if err := r.Patch(ctx, &existingPod, clients.Patch(apply), client.ForceOwnership, client.FieldOwner("shadow-pod")); err != nil {
+				klog.Errorf("unable to update pod %q: %v", klog.KObj(&existingPod), err)
+				return ctrl.Result{}, err
+			}
+
+			klog.Infof("updated pod %q with success.", klog.KObj(&existingPod))
+		} else {
+			klog.Infof("pod %q found running, skipping update because labels and annotations did not change", klog.KObj(&existingPod))
+		}
+
+		return ctrl.Result{}, nil
+	}
+
+	// ...Else, a brand new Pod must be created, based on the shadowpod.
+	newPod := corev1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:        nsName.Name,
 			Namespace:   nsName.Namespace,
-			Labels:      labels.Merge(shadowPod.Labels, labels.Set{consts.ManagedByLabelKey: consts.ManagedByShadowPodValue}),
+			Labels:      shadowPod.Labels,
 			Annotations: shadowPod.Annotations,
 		},
 		Spec: shadowPod.Spec.Pod,
 	}
 
-	utilruntime.Must(ctrl.SetControllerReference(&shadowPod, &pod, r.Scheme))
-
-	if err := r.Get(ctx, nsName, &pod); err == nil {
-		klog.V(4).Infof("skip: pod %q already running", klog.KObj(&pod))
-		return ctrl.Result{}, nil
-	}
+	utilruntime.Must(ctrl.SetControllerReference(&shadowPod, &newPod, r.Scheme))
 
-	if err := r.Create(ctx, &pod); err != nil {
+	if err := r.Create(ctx, &newPod); err != nil {
```

Comment:
You should specify the same FieldOwner as that used for the Patch operation, otherwise it won't work correctly.